integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOCILoadBalancer.java

```diff
@@ -0,0 +1,321 @@
+
+    logger.info("Get a unique namespace for WebLogic domain1");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domain1Namespace = namespaces.get(1);
+
+    logger.info("install and verify operator");
+    installAndVerifyOperator(opNamespace, domain1Namespace);
+  }
+
+  @AfterAll
+  public void tearDownAll() {
+    Kubernetes.deleteService(OCI_LB_NAME, domain1Namespace);
+  }
+
+  /**
+   * Test covers basic functionality for OCI LoadBalancer .
+   * Create omain and  OCI LoadBalancer.
+   * Check that application is accessabale via OCI LoadBalancer
+   */
+  @Test
+  @DisplayName("Test the sample-app app can be accessed"
+      + " from all managed servers in the domain through OCI Load Balancer.")
+  public void testOCILB() throws Exception {
+
+    // create and verify one cluster mii domain
+    logger.info("Create domain and verify that it's running");
+    String miiImage1 = createAndVerifyMiiImage(MODEL_DIR + "/model-singleclusterdomain-sampleapp-wls.yaml");
+    // create docker registry secret to pull the image from registry
+    // this secret is used only for non-kind cluster
+    logger.info("Create docker registry secret in namespace {0}", domain1Namespace);
+    createOcirRepoSecret(domain1Namespace);
+    createAndVerifyDomain(miiImage1, domain1Namespace, domain1Uid);
+    int clusterHttpPort = 8001;
+
+    assertDoesNotThrow(() -> installAndVerifyOCILoadBalancer(domain1Namespace,
+        clusterHttpPort, cluster1Name, domain1Uid, OCI_LB_NAME),
+        "Installation of OCI Load Balancer failed");
+    loadBalancerIP = getLoadBalancerIP(domain1Namespace,OCI_LB_NAME);
+    assertNotNull(loadBalancerIP, " External IP for Load Balancer is undefined");
+    logger.info(" LoadBalancer IP is " + loadBalancerIP);
+    verifyWebAppAccessThroughOCILB(loadBalancerIP, 2, clusterHttpPort);
+  }
+
+  /** Retreive external IP from OCI LoadBalancer.
+   *
+   */
+  private static String getLoadBalancerIP(String namespace, String lbName) throws Exception {
+    Map<String, String> labels = new HashMap<>();
+    labels.put("loadbalancer", lbName);
+    V1Service service = getService(lbName, labels, namespace);
+    assertNotNull(service, "Can't find service with name " + lbName);
+    logger.info("Found service with name {0} in {1} namespace ", lbName, namespace);
+    List<V1LoadBalancerIngress> ingress = service.getStatus().getLoadBalancer().getIngress();
+    if (ingress != null) {
+      logger.info("LoadBalancer Ingress " + ingress.toString());
+      V1LoadBalancerIngress lbIng = ingress.stream().filter(c ->
+          !c.getIp().equals("pending")
+      ).findAny().orElse(null);
+      if (lbIng != null) {
+        logger.info("OCI LoadBalancer is created with external ip" + lbIng.getIp());
+        return lbIng.getIp();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Create mii image with sample-app application.
+   */
+  private static String createAndVerifyMiiImage(String modelFile) {
```

Comment:
This image is created by default while running integration test. Look at the ItMiiDomain class. No need to define this method.

Reply:
done